### DISCUSSION TITLE
Dúvida layout tabela

### DISCUSSION THREAD

**Discussion by ifdiego** [frontendbr/forum#2100] [Perguntas e Respostas]
Oi galera,

Estou com uma dúvida referente ao layout de uma tabela.
É possível dividir uma tabela em duas dinamicamente e colocar outro elemento no meio?

Para visualização, o layout que estou fazendo é basicamente de uma [Danfe/NFE](https://upload.wikimedia.org/wikipedia/commons/9/97/Nfe.png).

Os items da tabela de NFE são carregados dinamicamente pelo backend.
A quantidade e o tamanho das linhas eu não posso deixar com tamanho fixo.

O elemento que ficará entre as tabelas é o ISSQN da imagem, é um footer com **position: fixed; bottom: 0;**
A tabela com os items da NFE só será dividida se a quantidade de items sobrepor ao footer.
E o layout será apenas para impressão.

**Comment by vitebo** [comment]
a gente não consegue pensar nelas como duas tabelas separadas? por exemplo:

```html
<table>
  ...
</table>
<div>
  Conteúdo opcional
</div>
<table>
  ...
</table>
```

**Reply by ifdiego** [reply]
Também pensei nisso, mas não consegui encontrar um jeito de separar as duas tabelas.

A div com o conteúdo opcional tem uma altura fixa de 170 pixels, mas os items da primeira tabela não, tanto em quantidade de items quanto o tamanho das linhas podem variar bastante.

Eu teria que pegar pela altura, quebrar em duas tabelas se os items alcançarem a altura do elemento fixo.
Mas não sei como fazer isso.

**Comment by ifdiego** [comment]
Talvez eu tenha encontrado uma possível solução utilizando as tags **thead** e **tfoot**.
Porém, não testei o suficiente pra saber ao certo se realmente funcionará.

A ideia é criar um footer com o tamanho e conteúdo fixo que preciso.
Encontrei a dica no [Codepen](https://codepen.io/srees-evd/full/wvvRmLw).

Aviso aqui, caso consiga.
